payment_paybox/models/inherited_payment_transaction.py

```diff
@@ -58,6 +58,28 @@ def _paybox_form_get_tx_from_data(self, data):
 
             _logger.info(error_msg)
             raise ValidationError(error_msg)
+        elif transaction.acquirer_reference and data.get('transaction') != transaction.acquirer_reference:
+            # In Paybox, if an error occurred due to 3DSecure authentication for example, Paybox store the transaction
+            # as an error and call IPN route. It means that the transaction in Odoo is update, set the state as 'error',
+            # and store the acquirer reference in it. But Paybox does not redirect to Odoo after this error, the user is
+            # allowed to retry the payment, but it generates a new reference in Paybox. So when the user paid, he calls
+            # the IPN route with another reference, and we get an error where it's impossible to update the transaction
+            # because the references are different.
+            # We first search for an existing transaction whichgenerate a new transaction with a reference updated, to have the same history as Paybox.
+            reference_transaction = transaction.reference.split('x')
```

Comment:
Modifications optionnelles

```suggestion
        elif transaction.acquirer_reference and data.get('transaction') != transaction.acquirer_reference:
            # In Paybox, if an error occurs due to 3DSecure authentication for example, Paybox store the transaction
            # as an error and call IPN route. It means that the transaction in Odoo is updated, the state is set as 'error',
            # and store the acquirer reference in it. But Paybox does not redirect to Odoo after this error, the user is
            # allowed to retry the payment, if so, it generates a new reference in Paybox. So when the user pay, the 
            # IPN route is called with another reference, and we get an error where it's impossible to update the transaction
            # because the references are different.
            # We first search for an existing transaction which generate a new transaction with a reference updated, to have the same history as Paybox.
            reference_transaction = transaction.reference.split('x')
```